src/app/components/UploadImage/UploadImage.tsx

```diff
@@ -0,0 +1,44 @@
+import styles from './UploadImage.module.css';
+import UploadImageIcon from '../../components/UploadImage/UploadImageIcon.svg';
+import { ChangeEvent, useState } from 'react';
+
+type ImageUploadProps = {
+  onUpload: (url: string) => void;
+};
+
+function UploadImage({ onUpload }: ImageUploadProps): JSX.Element {
+  const [imageSelected, setImageSelected] = useState<File | null>(null);
+
+  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    if (!event.target.files || event.target.files.length === 0) {
+      return;
+    }
+    setImageSelected(event.target.files[0]);
+
+    const formData = new FormData();
+    formData.append('file', imageSelected);
```

Comment:
You can not add a file (`imageSelected`) here, because it expects a `string` or `Blob` (see error message).
Maybe this is helpful?

https://developer.mozilla.org/en-US/docs/Web/API/URL/createObjectURL